lib/commands/doctor.js

```diff
@@ -139,7 +139,7 @@ class Doctor extends BaseCommand {
   async checkPing () {
     log.info('doctor', 'Pinging registry')
     try {
-      await ping({ ...this.npm.flatOptions, retry: false })
+      await ping({ ...this.npm.flatOptions, cache: false, retry: false })
```

Comment:
This should go directly into `lib/utils/ping.js` instead of us having to remember to pass it in everywhere.

Reply:
While we're in `lib/utils/ping.js` we should remove the `?write=true` since that doesn't actually address this bug.

Reply:
Makes sense.. Initially I kept it separate because it's being used in two different places and incase we intended to use this for something else, but yes I'll add test and make a central change in `utils/ping.js`.

Reply:
Removed `?write=true` in favour of `cache:false`

- `write=true` works for success response without disabling cache as it would sent options as `no-cache` in `make-fetch-happen` and forces it to revalidate the request, however when there is a ENOTFOUND error code it falls back to cached version of the response. ( from quick analysis )
- making cache: false will result in cache policy of `not storable`  making it new request and not the cached one.